sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -648,13 +649,80 @@ public override void Dispose()
             }
         }
 
+        private async Task<AmqpSession> CreateSessionIfNeededAsync(AmqpConnection connection, TimeSpan timeout, string transactionGroupKey)
+        {
+            AmqpSession session;
+            FaultTolerantAmqpObject<Controller> faultTolerantController = null;
+            if (transactionGroupKey != null)
+            {
+                if (TransactionGroups.ContainsKey(transactionGroupKey))
+                {
+                    return await TransactionGroups[transactionGroupKey].Session.GetOrCreateAsync(timeout).ConfigureAwait(false);
+                }
+                else
+                {
+                    // each transaction group will have its own session and controller
+                    var faultTolerantSession = new FaultTolerantAmqpObject<AmqpSession>(
+                        async (timeout) =>
+                        {
+                            var stopWatch = ValueStopwatch.StartNew();
+                            AmqpSession session = await CreateAndOpenSessionAsync(connection, timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
+                            await CreateControllerAsync(session, timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
```

Comment:
I'm surprised this didn't get flagged by the analyzer for "don't ignore return values", which I seem to trip in these scenarios. You may want to consider an explicit discard to guard against that popping up in the future.
```suggestion
                            _ = await CreateControllerAsync(session, timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
```